### PR TITLE
[TOPI][RELAY][TENSORFLOW]Math ops added

### PR DESCRIPTION
**Pull request by siju-samuel**
Added relay/topi/tensorflow support for the following ops.

- Acos
- Acosh
- Asin
- Asinh
- Atanh
- Cosh
- Sinh

@FrozenGene @masahi Please help to review this PR.

### CODE REVIEW

**Comment by siju-samuel** [comment]
@masahi @FrozenGene @kazum Please help to review. TIA

File: python/tvm/relay/op/_tensor_grad.py [hunk 1]
```diff
@@ -116,12 +117,60 @@ def sinh_grad(orig, grad):
...
     return [grad * cosh(x)]
 
+
+@register_gradient("acos")
+def acos_grad(orig, grad):
+    """Returns [grad * -1/((1 - (x ^ 2)) ^ 1/2)]"""
+    x = orig.args[0]
+    a = const(2.0)
+    ones = ones_like(x)
+    return [grad * (-ones / sqrt(ones_like(x) - power(x, a)))]
```
Comment (kazum) [inline python/tvm/relay/op/_tensor_grad.py:127]: ones_like(x) => ones

File: python/tvm/relay/op/_tensor_grad.py [hunk 2]
```diff
@@ -116,12 +117,60 @@ def sinh_grad(orig, grad):
...
     x = orig.args[0]
     a = const(2.0)
-    return [grad * ones_like(x) / (ones_like(x) + power(x, a))]
+    ones = ones_like(x)
+    return [grad * ones / (ones + power(x, a))]
+
+
+@register_gradient("atanh")
+def atanh_grad(orig, grad):
+    """Returns xx[grad * 1 / (1 - x ^ 2)]"""
```
Comment (kazum) [inline python/tvm/relay/op/_tensor_grad.py:168]: Remove `xx`.

File: python/tvm/relay/op/_tensor_grad.py [hunk 3]
```diff
@@ -116,12 +117,60 @@ def sinh_grad(orig, grad):
...
+    return [grad * (-ones / sqrt(ones_like(x) - power(x, a)))]
+
+
+@register_gradient("acosh")
+def acosh_grad(orig, grad):
+    """Returns [grad * 1/((x - 1) ^ 1/2 * (x + 1) ^ 1/2)]"""
+    x = orig.args[0]
+    a = const(2.0)
+    ones = ones_like(x)
+    return [grad * ones / sqrt(power(x, a) - ones)]
```
Comment (kazum) [inline python/tvm/relay/op/_tensor_grad.py:136]: I think all the `power(x, const(2.0))` can be replaced with `x * x`.

File: tests/python/relay/test_op_grad_level1.py [hunk 4]
```diff
@@ -69,7 +69,12 @@ def check_single_op(opfunc, ref):
                         (tvm.relay.log2, lambda x: 1 / (np.log(2) * x)),
                         (tvm.relay.log10, lambda x: 1 / (np.log(10) * x)),
                         (tvm.relay.cosh, lambda x: -1.0 * np.sinh(x)),
```
Comment (kazum) [inline tests/python/relay/test_op_grad_level1.py:71]: I think cosh'(x) is not -sinh(x) but sinh(x).

File: python/tvm/relay/op/_tensor_grad.py [hunk 5]
```diff
@@ -110,18 +110,66 @@ def sin_grad(orig, grad):
...
     return [grad * cosh(x)]
 
+
+@register_gradient("acos")
+def acos_grad(orig, grad):
+    """Returns [grad * -1/((1 - (x ^ 2)) ^ 1/2)]"""
+    x = orig.args[0]
+    a = const(2.0)
+    ones = ones_like(x)
+    return [grad * (-ones / sqrt(ones - power(x, a)))]
```
Comment (kazum) [inline python/tvm/relay/op/_tensor_grad.py:127]: Replace `power(x,a)` with `x*x`, and remove `a = const(2.0)`.

File: python/tvm/relay/op/_tensor_grad.py [hunk 6]
```diff
@@ -110,18 +110,66 @@ def sin_grad(orig, grad):
...
+    return [grad * ones / sqrt((x * x) - ones)]
+
+
+@register_gradient("asin")
+def asin_grad(orig, grad):
+    """Returns [grad * 1/((1 - (x ^ 2)) ^ (1/2))]"""
+    x = orig.args[0]
+    a = const(2.0)
+    ones = ones_like(x)
+    return [grad * ones / sqrt(ones - power(x, a))]
```
Comment (kazum) [inline python/tvm/relay/op/_tensor_grad.py:144]: Same here.

File: python/tvm/relay/op/_tensor_grad.py [hunk 7]
```diff
@@ -110,18 +110,66 @@ def sin_grad(orig, grad):
...
+
+
+@register_gradient("asinh")
+def asinh_grad(orig, grad):
+    """Returns [grad * 1/((1 + (x ^ 2)) ^ (1/2))]"""
+    x = orig.args[0]
+    a = const(2.0)
+    ones = ones_like(x)
+    return [grad * ones / sqrt(ones + power(x, a))]
```
Comment (kazum) [inline python/tvm/relay/op/_tensor_grad.py:153]: Same here.

File: python/tvm/relay/op/_tensor_grad.py [hunk 8]
```diff
@@ -110,18 +110,66 @@ def sin_grad(orig, grad):
...
+
+
 @register_gradient("atan")
 def atan_grad(orig, grad):
     """Returns [grad * 1 / (1 + x ^ 2)]"""
     x = orig.args[0]
     a = const(2.0)
-    return [grad * ones_like(x) / (ones_like(x) + power(x, a))]
+    ones = ones_like(x)
+    return [grad * ones / (ones + power(x, a))]
```
Comment (kazum) [inline python/tvm/relay/op/_tensor_grad.py:162]: Same here.

File: python/tvm/relay/op/_tensor_grad.py [hunk 9]
```diff
@@ -110,18 +110,66 @@ def sin_grad(orig, grad):
...
+
+
+@register_gradient("atanh")
+def atanh_grad(orig, grad):
+    """Returns [grad * 1 / (1 - x ^ 2)]"""
+    x = orig.args[0]
+    a = const(2.0)
+    ones = ones_like(x)
+    return [grad * ones / (ones - power(x, a))]
```
Comment (kazum) [inline python/tvm/relay/op/_tensor_grad.py:171]: Same here.

**Comment by kazum** [comment]
Thanks @siju-samuel !

**Pull request merged**